tests/frontend/test_program.py

```diff
@@ -672,6 +672,81 @@ class DummyCircuit(Compiler):
                 compiler=DummyCircuit(),
             )
 
+    def test_missing_layout(self):
+        """Test that an error is raised if the device spec is missing a layout when compiled"""
+        device_dict = {
+            "target": None,
+            "layout": None,
+            "modes": 2,
+            "compiler": ["DummyCompiler"],
+            "gate_parameters": {
+                "squeezing_amplitude_0": [0, 1],
+            },
+        }
+
+        class DummyCircuit(Compiler):
```

Comment:
Minor point, but is this a typo?
```suggestion
        class DummyCompiler(Compiler):
```

Reply:
Likely, I believe it's been there for some time, but haven't been updated (due to it being quite minor).